extensions/jms/src/main/java/org/mule/extensions/jms/api/source/JmsListener.java

```diff
@@ -172,23 +178,35 @@ public void onStop() {
     if (LOGGER.isDebugEnabled()) {
       LOGGER.debug("Stopping JMSSubscriber source");
     }
+
+    jmsLock.unlock();
+
+    try {
+      if (ackMode.equals(AUTO) || ackMode.equals(DUPS_OK) || ackMode.equals(MANUAL)) {
```

Comment:
if we don't want to recover only for NONE, then is more explicit to check for that instead all the other options